crates/volta-core/src/tool/package_global/mod.rs

```diff
@@ -31,7 +37,16 @@ impl Package {
         })
     }
 
-    pub fn persist_install(self) -> Fallible<()> {
+    pub fn complete_install(self, image: &Image) -> Fallible<PackageManifest> {
+        let manifest = self.parse_manifest()?;
+
+        self.write_config_and_shims(&manifest, &image)?;
+        self.persist_install()?;
```

Comment:
Seems like `persist_install()` should be called before `write_config_and_shims()`? So that for the case where there is an error with `persist_install()`, there won't be any shims or config written for a package that doesn't exist in the expected location. For this current order, if there is an error with `persist_install()`, you can have shims and config that exist, but the package may not be there.

Reply:
I went back and forth on this, since the reverse could happen as well (the directory gets overwritten and then writing the configs fails). But thinking about it more, for a new install, in that case at least we don't leave things with broken shims. Updated.